cub/cub/device/dispatch/dispatch_reduce_nondeterministic.cuh

```diff
@@ -0,0 +1,312 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+/**
+ * @file cub::DeviceReduce provides device-wide, parallel operations for computing a reduction
+ *       across a sequence of data items residing within device-accessible memory.
```

Comment:
Q: Is this comment still correct?

It's also wrong doxygen syntax: It needs a line break after @file:
```suggestion
 * @file
 * cub::DeviceReduce provides device-wide, parallel operations for computing a reduction
 *       across a sequence of data items residing within device-accessible memory.
```
Also, please use RST comments `//!`